cmd/bee/cmd/start.go

```diff
@@ -420,36 +412,38 @@ func (c *command) configureSigner(cmd *cobra.Command, logger logging.Logger) (co
 	}, nil
 }
 
-func parseNetworks(main bool, networkID uint64) (uint64, error) {
-	if main && networkID != 1 {
-		return 0, errors.New("provided network ID does not match mainnet")
-	}
-
-	return networkID, nil
+type mainnetCfg struct {
+	networkID uint64
+	bootnodes []string
+	blockTime uint64
 }
 
-func parseBootnodes(log logging.Logger, main bool, networkID uint64, bootnodes []string) []string {
-	if len(bootnodes) > 0 {
-		return bootnodes // use provided values
-	}
-
-	if main {
-		return []string{"/dnsaddr/mainnet.ethswarm.org"}
-	}
+func getMainnectConfig(c *viper.Viper, log logging.Logger) (*mainnetCfg, error) {
```

Comment:
I think rather than using the config flags in here this should simply build the defaults for a given network id. then some other code (independent of a specific network) can override values with the config flags.